stdlib/socket/src/main/java/org/ballerinalang/stdlib/socket/tcp/ReadPendingSocketMap.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.stdlib.socket.tcp;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This map will hold all the read pending action that coming through
+ * {@link org.ballerinalang.stdlib.socket.endpoint.tcp.client.Read} action until new content ready.
+ *
```

Comment:
This map will hold all the pending read actions that are coming through the
 * {@link org.ballerinalang.stdlib.socket.endpoint.tcp.client.Read} action until the new content is ready.